### PR TITLE
 support transfer with signer 

### PR DESCRIPTION
**Pull request by jazz-cb**
### What changed? Why?
Support transfer with ServerSigner.
1. Change logic in status to return what is in the model
2. Add a reload method that updates the model with what is in backend
3. Update wait! to use reload instead of relying on status

```
[6] pry(main)> t = w.transfer(0.0001, :eth, w2).wait!
=> Coinbase::Transfer{transfer_id: 'f199e5f6-b18f-4aca-b4b5-4075c0feabb1', network_id: 'base_sepolia', from_address_id: '0x88DBBf024Ef56137bc688FAed7AffD20474d80B8', destination_address_id: '0x7b422c56A66e0A8c789Ac0e29313b127e9CDd0e0', asset_id: 'eth', amount: '0.1e-3', transaction_hash: '0x1fa2d646ded18f69550207f8e3979ec031361883e5e95c55d7911464da4d694f', transaction_link: 'https://sepolia.basescan.org/tx/0x1fa2d646ded18f69550207f8e3979ec031361883e5e95c55d7911464da4d694f', status: 'complete'}
```

### CODE REVIEW

File: lib/coinbase/transfer.rb [hunk 1]
```diff
@@ -166,7 +168,15 @@ def wait!(interval_seconds = 0.2, timeout_seconds = 10)
...
       loop do
-        return self if status == Status::COMPLETE || status == Status::FAILED
+        update = Coinbase.call_api do
+          transfers_api.get_transfer(wallet_id, from_address_id, id)
+        end
+
+        if update.status.to_s == Status::COMPLETE.to_s || update.status == Status::FAILED.to_s
+          self.transaction_hash = update.transaction_hash
+          self.status = update.status
+          return self
```
Comment (alex-stone) [inline lib/coinbase/transfer.rb:178]: Can we just update the `model` and then have `transaction_hash` and `status` be derived from that?

Comment (jazz-cb) [reply lib/coinbase/transfer.rb]: this is how the latest commit handles the update

File: lib/coinbase/transfer.rb [hunk 2]
```diff
@@ -166,7 +154,9 @@ def wait!(interval_seconds = 0.2, timeout_seconds = 10)
       start_time = Time.now
 
       loop do
-        return self if status == Status::COMPLETE || status == Status::FAILED
+        reload
+
+        return self if status == Status::COMPLETE.to_s || status == Status::FAILED.to_s
```
[the remark this answers is not in the document]
Comment (alex-stone) [reply lib/coinbase/transfer.rb:159]: If we have our own types here, should we convert from string to actual type? Ideally we wouldn't have these have to be cast if these are things that developers can leverage

File: spec/unit/coinbase/address_spec.rb [hunk 3]
```diff
@@ -201,6 +201,7 @@
     let(:transfer) do
       double('Transfer', transaction: transaction, id: transfer_id)
     end
+    let(:configuration) { double('Coinbase::Configuration', use_server_signer: use_server_signer, api_client: nil) }
```
[the remark this answers is not in the document]
Comment (jazz-cb) [reply spec/unit/coinbase/address_spec.rb:204]: fixed

File: lib/coinbase/address.rb [hunk 4]
```diff
@@ -108,6 +108,9 @@ def transfer(amount, asset_id, destination)
         transfers_api.create_transfer(wallet_id, id, create_transfer_request)
       end
 
+      # Return here if ServerSigner is expected to sign the transfer.
+      return Coinbase::Transfer.new(transfer_model) if Coinbase.use_server_signer?
```
[the remark this answers is not in the document]
Comment (jazz-cb) [reply lib/coinbase/address.rb:112]: yes, we can do this as a followup

[the remark this answers is not in the document]
Comment (alex-stone) [reply lib/coinbase/address.rb:112]: https://github.com/coinbase/coinbase-sdk-ruby/pull/57

[the remark this answers is not in the document]
Comment (alex-stone) [reply lib/coinbase/address.rb:112]: Doing this as trade, etc... will be using these same paradigms!

**Review by alex-stone — APPROVED** [review]

**Pull request merged**